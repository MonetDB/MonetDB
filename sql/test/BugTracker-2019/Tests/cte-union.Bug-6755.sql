start transaction;
CREATE TABLE ontime (
        "Year"       SMALLINT,
        "Month"      TINYINT,
        "DayofMonth" TINYINT,
        "Carrier"    CHAR(2),
        "CRSDepTime" DECIMAL(8,2),
        "ArrDelay"   DECIMAL(8,2)
);
CREATE TABLE tmp (
        "Hour" TINYINT, 
        "PredictedArrDelay" DECIMAL(8,2) DEFAULT 0.0
);
INSERT INTO tmp ("Hour")
VALUES
    (0), (1), (2), (3), (4), (5), 
    (6), (7), (8), (9), (10), (11), 
    (12), (13), (14), (15), (16), (17), 
    (18), (19), (20), (21), (22), (23);

INSERT INTO ontime VALUES (2001, 9, 2, 'AA', 900.00, -6.00);
ALTER TABLE "ontime" SET READ ONLY;
ANALYZE sys.ontime;
WITH t1 AS (
    SELECT "Carrier", CAST (FLOOR("CRSDepTime"%2400/100) AS INT) AS "Hour", 
           CAST(AVG("ArrDelay") AS DECIMAL(8,2)) AS "PredictedArrDelay"
    FROM ontime
    WHERE "Year" = 2007 AND "Month" = 10 AND "DayofMonth" = 24
    GROUP BY "Carrier", "Hour"
),
t2 AS (
    SELECT t."Carrier", tmp.*
    FROM tmp, (SELECT DISTINCT "Carrier" FROM t1) AS t
)
SELECT "Carrier", "Hour", CAST(SUM("PredictedArrDelay") AS DECIMAL(18,2))
FROM (SELECT * FROM t1 UNION SELECT * FROM t2) AS t
GROUP BY "Carrier", "Hour"
ORDER BY "Carrier", "Hour";
rollback;
