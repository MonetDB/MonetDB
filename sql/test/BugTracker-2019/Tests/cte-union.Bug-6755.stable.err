stderr of test 'cte-union.Bug-6755` in directory 'sql/test/BugTracker-2019` itself:


# 22:24:23 >  
# 22:24:23 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=38598" "--set" "mapi_usock=/var/tmp/mtest-21682/.s.monetdb.38598" "--forcemito" "--dbpath=/home/niels/scratch/rc-nov2019/Linux-x86_64/var/MonetDB/mTests_sql_test_BugTracker-2019" "--set" "embedded_c=true"
# 22:24:23 >  

# builtin opt 	gdk_dbpath = /home/niels/scratch/rc-nov2019/Linux-x86_64/var/monetdb5/dbfarm/demo
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_ipv6 = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 38598
# cmdline opt 	mapi_usock = /var/tmp/mtest-21682/.s.monetdb.38598
# cmdline opt 	gdk_dbpath = /home/niels/scratch/rc-nov2019/Linux-x86_64/var/MonetDB/mTests_sql_test_BugTracker-2019
# cmdline opt 	embedded_c = true

# 22:24:23 >  
# 22:24:23 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-21682" "--port=38598"
# 22:24:23 >  


# 22:24:24 >  
# 22:24:24 >  "Done."
# 22:24:24 >  

