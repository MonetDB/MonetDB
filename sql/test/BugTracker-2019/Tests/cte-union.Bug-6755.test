statement ok
start transaction

statement ok
CREATE TABLE ontime (
        "Year"       SMALLINT,
        "Month"      TINYINT,
        "DayofMonth" TINYINT,
        "Carrier"    CHAR(2),
        "CRSDepTime" DECIMAL(8,2),
        "ArrDelay"   DECIMAL(8,2)
)

statement ok
CREATE TABLE tmp (
        "Hour" TINYINT,
        "PredictedArrDelay" DECIMAL(8,2) DEFAULT 0.0
)

statement ok
INSERT INTO tmp ("Hour")
VALUES
    (0), (1), (2), (3), (4), (5),
    (6), (7), (8), (9), (10), (11),
    (12), (13), (14), (15), (16), (17),
    (18), (19), (20), (21), (22), (23)

statement ok
INSERT INTO ontime VALUES (2001, 9, 2, 'AA', 900.00, -6.00)

statement ok
ALTER TABLE "ontime" SET READ ONLY

statement ok
ANALYZE sys.ontime

query TIR rowsort
WITH t1 AS (
    SELECT "Carrier", CAST (FLOOR("CRSDepTime"%2400/100) AS INT) AS "Hour",
           CAST(AVG("ArrDelay") AS DECIMAL(8,2)) AS "PredictedArrDelay"
    FROM ontime
    WHERE "Year" = 2007 AND "Month" = 10 AND "DayofMonth" = 24
    GROUP BY "Carrier", "Hour"
),
t2 AS (
    SELECT t."Carrier", tmp.*
    FROM tmp, (SELECT DISTINCT "Carrier" FROM t1) AS t
)
SELECT "Carrier", "Hour", CAST(SUM("PredictedArrDelay") AS DECIMAL(18,2))
FROM (SELECT * FROM t1 UNION SELECT * FROM t2) AS t
GROUP BY "Carrier", "Hour"
ORDER BY "Carrier", "Hour"
----

statement ok
rollback

