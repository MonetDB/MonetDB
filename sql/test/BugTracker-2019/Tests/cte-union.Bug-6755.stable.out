stdout of test 'cte-union.Bug-6755` in directory 'sql/test/BugTracker-2019` itself:


# 22:24:23 >  
# 22:24:23 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=38598" "--set" "mapi_usock=/var/tmp/mtest-21682/.s.monetdb.38598" "--forcemito" "--dbpath=/home/niels/scratch/rc-nov2019/Linux-x86_64/var/MonetDB/mTests_sql_test_BugTracker-2019" "--set" "embedded_c=true"
# 22:24:23 >  

# MonetDB 5 server v11.35.0 (hg id: d2a25d02ca50+)
# This is an unreleased version
# Serving database 'mTests_sql_test_BugTracker-2019', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 15.384 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2020 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://xps13:38598/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-21682/.s.monetdb.38598
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

# 22:24:23 >  
# 22:24:23 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-21682" "--port=38598"
# 22:24:23 >  

#start transaction;
#CREATE TABLE ontime (
#        "Year"       SMALLINT,
#        "Month"      TINYINT,
#        "DayofMonth" TINYINT,
#        "Carrier"    CHAR(2),
#        "CRSDepTime" DECIMAL(8,2),
#        "ArrDelay"   DECIMAL(8,2)
#);
#CREATE TABLE tmp (
#        "Hour" TINYINT, 
#        "PredictedArrDelay" DECIMAL(8,2) DEFAULT 0.0
#);
#INSERT INTO tmp ("Hour")
#VALUES
#    (0), (1), (2), (3), (4), (5), 
#    (6), (7), (8), (9), (10), (11), 
#    (12), (13), (14), (15), (16), (17), 
#    (18), (19), (20), (21), (22), (23);
[ 24	]
#INSERT INTO ontime VALUES (2001, 9, 2, 'AA', 900.00, -6.00);
[ 1	]
#ALTER TABLE "ontime" SET READ ONLY;
#WITH t1 AS (
#    SELECT "Carrier", CAST (FLOOR("CRSDepTime"%2400/100) AS INT) AS "Hour", 
#           CAST(AVG("ArrDelay") AS DECIMAL(8,2)) AS "PredictedArrDelay"
#    FROM ontime
#    WHERE "Year" = 2007 AND "Month" = 10 AND "DayofMonth" = 24
#    GROUP BY "Carrier", "Hour"
#),
#t2 AS (
#    SELECT t."Carrier", tmp.*
#    FROM tmp, (SELECT DISTINCT "Carrier" FROM t1) AS t
#)
#SELECT "Carrier", "Hour", SUM("PredictedArrDelay")
#FROM (SELECT * FROM t1 UNION SELECT * FROM t2) AS t
#GROUP BY "Carrier", "Hour"
% sys.t,	sys.t,	sys.%12 # table_name
% Carrier,	Hour,	%12 # name
% char,	int,	decimal # type
% 2,	1,	20 # length
#rollback;

# 22:24:24 >  
# 22:24:24 >  "Done."
# 22:24:24 >  

